### ISSUE TITLE
[BUG] Crashing when selecting password file using external repo

### ISSUE THREAD

**Issue by nitro322** [android-password-store/Android-Password-Store#600]
**Describe the bug**
Note: This follows on the fix for bug #593.  I'm using the 1.3.4 snapshot dated 2019-12-15 15:53 to test.

I can now select the .password-store repo synced to my phone by nextcloud and successfully browse the files, but open opening any of the password files the app instantly crashes.  Behavior is repeatable for any selected GPG password file.

If I use OpenKeychain directly to decrypt one of the password files, it works as expected, so I'm relatively certain the GPG password files are not corrupt and I have the correct key specified.

Sorry to make a nuisance of myself, but I noticed you have the 1.3.4 release planned for next week, so I wanted to get this posted ASAP in the hope of getting it resolved before then.

**To Reproduce**
1. Configure Password Store to use external repository as described in bug #593 
2. Select OpenKeychain as the OpenPGP Providor
3. Select your OpenPGP Key Id
4. Return to main screen
5. Select password file and observe crash
6. Re-launch app, select password file, and observe repeatable crash

**Expected behavior**
I'd expect to be prompted for my passphrase, then shown the password

**Device information (please complete the following information):**
 - Device: Pixel 3a
 - OS: stock ROM Android 10
 - App version: 1.3.4 snapshot dated 2019-12-15 15:53
 - OpenKeychain version: 5.4

**Additional context**
logcat output:

```
12-15 20:21:25.952   838 18173 I WifiHAL : event received NL80211_CMD_VENDOR, vendor_id = 0x1374, subcmd = 0xd
12-15 20:21:27.456 21600 21600 D AndroidRuntime: Shutting down VM
12-15 20:21:27.456 21600 21600 E AndroidRuntime: FATAL EXCEPTION: main
12-15 20:21:27.456 21600 21600 E AndroidRuntime: Process: com.zeapo.pwdstore, PID: 21600
12-15 20:21:27.456 21600 21600 E AndroidRuntime: java.lang.Error: java.lang.InstantiationException: java.lang.Class<z.b.a.i.a> cannot be instantiated
12-15 20:21:27.456 21600 21600 E AndroidRuntime:        at z.b.a.l.a.a(Unknown Source:53)
12-15 20:21:27.456 21600 21600 E AndroidRuntime:        at z.b.a.i.a.a(:2)
12-15 20:21:27.456 21600 21600 E AndroidRuntime:        at z.b.a.a.p.call(:9)
12-15 20:21:27.456 21600 21600 E AndroidRuntime:        at com.zeapo.pwdstore.PasswordStore.a(:10)
12-15 20:21:27.456 21600 21600 E AndroidRuntime:        at v.d.a.b.a(:30)
12-15 20:21:27.456 21600 21600 E AndroidRuntime:        at m.onClick(java-style lambda group:30)
12-15 20:21:27.456 21600 21600 E AndroidRuntime:        at android.view.View.performClick(View.java:7140)
12-15 20:21:27.456 21600 21600 E AndroidRuntime:        at android.view.View.performClickInternal(View.java:7117)
12-15 20:21:27.456 21600 21600 E AndroidRuntime:        at android.view.View.access$3500(View.java:801)
12-15 20:21:27.456 21600 21600 E AndroidRuntime:        at android.view.View$PerformClick.run(View.java:27351)
12-15 20:21:27.456 21600 21600 E AndroidRuntime:        at android.os.Handler.handleCallback(Handler.java:883)
12-15 20:21:27.456 21600 21600 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:100)
12-15 20:21:27.456 21600 21600 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:214)
12-15 20:21:27.456 21600 21600 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:7356)
12-15 20:21:27.456 21600 21600 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
12-15 20:21:27.456 21600 21600 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
12-15 20:21:27.456 21600 21600 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
12-15 20:21:27.456 21600 21600 E AndroidRuntime: Caused by: java.lang.InstantiationException: java.lang.Class<z.b.a.i.a> cannot be instantiated
12-15 20:21:27.456 21600 21600 E AndroidRuntime:        at java.lang.Class.newInstance(Native Method)
12-15 20:21:27.456 21600 21600 E AndroidRuntime:        at z.b.a.l.a.a(Unknown Source:33)
12-15 20:21:27.456 21600 21600 E AndroidRuntime:        ... 16 more
12-15 20:21:27.460  1280  1932 W ActivityTaskManager:   Force finishing activity com.zeapo.pwdstore/.PasswordStore
12-15 20:21:27.460  1280 21641 I DropBoxManagerService: add tag=data_app_crash isTagEnabled=true flags=0x2
12-15 20:21:27.469  1280  1345 I ActivityManager: Showing crash dialog for package com.zeapo.pwdstore u0
12-15 20:21:27.470  1280  1367 W BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
12-15 20:21:27.470  1280  1367 W BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
12-15 20:21:27.493   767   767 I netd    : bandwidthAddNiceApp(10237) <0.05ms>
12-15 20:21:27.511  1280  5540 E libc    : Access denied finding property "vendor.debug.egl.changepixelformat"
12-15 20:21:27.537  1280  5540 E libc    : Access denied finding property "vendor.debug.egl.swapinterval"
12-15 20:21:27.542  1280  5540 E libc    : Access denied finding property "vendor.debug.egl.swapinterval"
12-15 20:21:27.961   838 18173 I WifiHAL : event received NL80211_CMD_VENDOR, vendor_id = 0x1374, subcmd = 0xd
12-15 20:21:27.963  1280  1349 W ActivityTaskManager: Activity top resumed state loss timeout for ActivityRecord{bc4dc9 u0 com.zeapo.pwdstore/.PasswordStore t1022 f}
12-15 20:21:27.963  1280  1349 W ActivityTaskManager: Activity pause timeout for ActivityRecord{bc4dc9 u0 com.zeapo.pwdstore/.PasswordStore t1022 f}
12-15 20:21:28.046   767   767 I netd    : bandwidthRemoveNiceApp(10265) <0.05ms>
12-15 20:21:29.966   838 18173 I WifiHAL : event received NL80211_CMD_VENDOR, vendor_id = 0x1374, subcmd = 0xd
```

**Comment by msfjarvis** [comment]
New snapshot is up with this bug resolved.

**Comment by nitro322** [comment]
Thanks again for the quick reply, but the new build doesn't fix the issue for me.  I still have the same behavior.  I capture another logcat and get similar but slightly different output on the instantiation line:

```
12-16 01:18:25.501   830  1155 I ASH     : @ 82796.024: tmd2725: PROX sample: PDATA 8f
12-16 01:18:25.502   830  1155 I ASH     : @ 82796.025: tmd2725: PROX sample: NEAR
12-16 01:18:25.956  1176  1176 D AndroidRuntime: Shutting down VM
12-16 01:18:25.957  1176  1176 E AndroidRuntime: FATAL EXCEPTION: main
12-16 01:18:25.957  1176  1176 E AndroidRuntime: Process: com.zeapo.pwdstore, PID: 1176
12-16 01:18:25.957  1176  1176 E AndroidRuntime: java.lang.Error: java.lang.InstantiationException: java.lang.Class<org.eclipse.jgit.internal.JGitText> cannot be instantiated
12-16 01:18:25.957  1176  1176 E AndroidRuntime:        at org.eclipse.jgit.nls.GlobalBundleCache.lookupBundle(GlobalBundleCache.java:9)
12-16 01:18:25.957  1176  1176 E AndroidRuntime:        at org.eclipse.jgit.internal.JGitText.get(JGitText.java:4)
12-16 01:18:25.957  1176  1176 E AndroidRuntime:        at org.eclipse.jgit.api.LogCommand.call(LogCommand.java:26)
12-16 01:18:25.957  1176  1176 E AndroidRuntime:        at com.zeapo.pwdstore.PasswordStore.decryptPassword(PasswordStore.kt:20)
12-16 01:18:25.957  1176  1176 E AndroidRuntime:        at com.zeapo.pwdstore.PasswordFragment$onAttach$1.onFragmentInteraction(PasswordFragment.kt:43)
12-16 01:18:25.957  1176  1176 E AndroidRuntime:        at -$$LambdaGroup$js$mRUj0RD6kp4Vkw4ULifaVT4Ji4E.onClick(java-style lambda group:35)
12-16 01:18:25.957  1176  1176 E AndroidRuntime:        at android.view.View.performClick(View.java:7140)
12-16 01:18:25.957  1176  1176 E AndroidRuntime:        at android.view.View.performClickInternal(View.java:7117)
12-16 01:18:25.957  1176  1176 E AndroidRuntime:        at android.view.View.access$3500(View.java:801)
12-16 01:18:25.957  1176  1176 E AndroidRuntime:        at android.view.View$PerformClick.run(View.java:27351)
12-16 01:18:25.957  1176  1176 E AndroidRuntime:        at android.os.Handler.handleCallback(Handler.java:883)
12-16 01:18:25.957  1176  1176 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:100)
12-16 01:18:25.957  1176  1176 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:214)
12-16 01:18:25.957  1176  1176 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:7356)
12-16 01:18:25.957  1176  1176 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
12-16 01:18:25.957  1176  1176 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
12-16 01:18:25.957  1176  1176 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
12-16 01:18:25.957  1176  1176 E AndroidRuntime: Caused by: java.lang.InstantiationException: java.lang.Class<org.eclipse.jgit.internal.JGitText> cannot be instantiated
12-16 01:18:25.957  1176  1176 E AndroidRuntime:        at java.lang.Class.newInstance(Native Method)
12-16 01:18:25.957  1176  1176 E AndroidRuntime:        at org.eclipse.jgit.nls.GlobalBundleCache.lookupBundle(GlobalBundleCache.java:6)
12-16 01:18:25.957  1176  1176 E AndroidRuntime:        ... 16 more
12-16 01:18:25.961  1280  6723 W ActivityTaskManager:   Force finishing activity com.zeapo.pwdstore/.PasswordStore
12-16 01:18:25.961  1280  1434 I DropBoxManagerService: add tag=data_app_crash isTagEnabled=true flags=0x2
12-16 01:18:25.968  1280  1345 I ActivityManager: Showing crash dialog for package com.zeapo.pwdstore u0
12-16 01:18:25.969  1280  1345 E system_server: Invalid ID 0x00000000.
12-16 01:18:25.969  1280  1367 W BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
12-16 01:18:25.969  1280  1367 W BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
12-16 01:18:25.988   767   767 I netd    : bandwidthAddNiceApp(10237) <0.05ms>
12-16 01:18:26.011  1280  5540 E libc    : Access denied finding property "vendor.debug.egl.changepixelformat"
12-16 01:18:26.040  1280  5540 E libc    : Access denied finding property "vendor.debug.egl.swapinterval"
12-16 01:18:26.045  1280  5540 E libc    : Access denied finding property "vendor.debug.egl.swapinterval"
12-16 01:18:26.100   830  1155 I ASH     : @ 82796.627: tmd2725: PROX sample: PDATA 11
12-16 01:18:26.100   830  1155 I ASH     : @ 82796.628: tmd2725: PROX sample: FAR
12-16 01:18:26.465  1280  1349 W ActivityTaskManager: Activity top resumed state loss timeout for ActivityRecord{45d6ebc u0 com.zeapo.pwdstore/.PasswordStore t1074 f}
12-16 01:18:26.466  1280  1349 W ActivityTaskManager: Activity pause timeout for ActivityRecord{45d6ebc u0 com.zeapo.pwdstore/.PasswordStore t1074 f}
12-16 01:18:26.588   767   767 I netd    : bandwidthRemoveNiceApp(10269) <0.06ms>
12-16 01:18:26.744   838 18173 I WifiHAL : event received NL80211_CMD_VENDOR, vendor_id = 0x1374, subcmd = 0xd
```

**Comment by msfjarvis** [comment]
I cannot reproduce this on any of my devices :/

**Comment by nitro322** [comment]
Is there any additional data I can provide to assist with debugging?

Just FYI, I did both uninstall and delete the APK of the previous snapshot to be certain I'm using the latest version.

**Comment by msfjarvis** [comment]
> Is there any additional data I can provide to assist with debugging?
> 
> Just FYI, I did both uninstall and delete the APK of the previous snapshot to be certain I'm using the latest version.

I have simply no clue why this could be happening. I'm gonna re-open this issue for now and ask around to see if someone knows something I don't.

**Comment by msfjarvis** [comment]
Okay, I was able to reproduce this by trying to delete a file. Will investigate it later today.

**Comment by msfjarvis** [comment]
@nitro322 can you test [this](https://dl.msfjarvis.dev/.superhidden/APS-v1.3.3-87-g5cf8ef85ff09.apk) debug build and see if it fixes things?

**Comment by nitro322** [comment]
Works great!  Thanks so much.

**Comment by msfjarvis** [comment]
![image](https://user-images.githubusercontent.com/13348378/71559729-d58d9a00-2a87-11ea-9102-2ba822ff96ea.png)


TL;DR we'll be stuck targetting Android Pie until the v2 rewrite comes around where we can finally do things properly.

**Comment by nitro322** [comment]
That's... ominous.  So just to make sure I understand, aside from this test build, no newer releases will work on my phone (or Android 10 devices in general) until version 2.0 is released? Understood if that's the case, just want to make sure I understand what you mean by targeting Pie.

**Comment by msfjarvis** [comment]
> That's... ominous. So just to make sure I understand, aside from this test build, no newer releases will work on my phone (or Android 10 devices in general) until version 2.0 is released? Understood if that's the case, just want to make sure I understand what you mean by targeting Pie.

Oh no, nothing that catastrophic. It just means it'll be harder to take benefit of any features from Android 10 that we'd want to in the future and Google Play will possibly stop accepting updates after Android 11 is released, unless we go back targetting Android 10. I'm confident that v2 will be ready to replace the current app within 2020 so it's not particularly worrisome if everything goes per plan.

**Comment by nitro322** [comment]
Oh, gotcha. Thanks for clarifying, and again, thanks a bunch for fixing this. Good luck with the migration efforts - looks like quite a bit of work.

**Comment by ulfi** [comment]
Info:
After the update of my App V1.3.3 from f-droid on my Xperia XZ2 from Androd 9 to 10. The program crashes during opening. If I delete the memory, the App works till the storage location will be mapped and I leave the settings dialog.

**Comment by msfjarvis** [comment]
We're aware, that's what this fixes.

**Comment by msfjarvis** [comment]
Closing since this is fixed on master
